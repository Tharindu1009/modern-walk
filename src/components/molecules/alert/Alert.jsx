import './styles/alert.scss';

function Alert({ title }) {
    return (
        <div data-testid="alert" className='alert'>
            {title}
        </div>
    )
}

export default Alert;