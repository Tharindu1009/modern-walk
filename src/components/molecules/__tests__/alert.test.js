import { render, cleanup } from "@testing-library/react";
import TestRenderer from 'react-test-renderer';
import Alert from "../alert/Alert";

describe('Alert', () => {
    afterEach(() => {
        cleanup();
    })

    it('should render alert component correctly', () => {
        const { getByTestId } = render(<Alert title="No data found." />);
        const alertElement = getByTestId('alert');
        expect(alertElement).toBeInTheDocument();
    });

    it('should display alert text content correctly', () => {
        const { getByTestId } = render(<Alert title="No data found." />);
        const alertElement = getByTestId('alert');
        expect(alertElement).toHaveTextContent('No data found.');
    });

    it('matches alert snapshot', () => {
        const tree = TestRenderer.create(<Alert title="No data found." />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});