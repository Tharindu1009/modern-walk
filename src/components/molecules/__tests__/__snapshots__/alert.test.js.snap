// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert matches alert snapshot 1`] = `
<div
  className="alert"
  data-testid="alert"
>
  No data found.
</div>
`;
